UnitTests/MeshTests.h

```diff
@@ -122,4 +123,42 @@ DEFINE_TEST_CASE(MeshWriteBinary)
         mesh[0].b.x == 0 && mesh[0].b.y == 1 && mesh[0].b.z == 0 &&
         mesh[0].c.x == 0 && mesh[0].c.y == 1 && mesh[0].c.z == -1
         ), "The written mesh is wrong!");
+}
+
+DEFINE_TEST_CASE(FoundationCreationFacetsModel)
+{
+    Mesh mesh;
+    mesh.read("../test_models/facets_155314.stl");
+    Mesh output_mesh = createFoundation(mesh, 6.00, 5.00);
+    auto facets = output_mesh.getFacets();
+    int i = 0;
+    TEST_ASSERT(facets[i].getNormal() == FPoint3D(0, 0, 1), "Normal of facets on top must be {0, 0, 1}");
+    int j = facets.size() - 1;
+    TEST_ASSERT(facets[j].getNormal() == FPoint3D(0, 0, -1), "Normal of facets on bottom must be {0, 0, -1}");
+    output_mesh.writeBinary("../test_models/facets_foundation.stl");
+}
+
+DEFINE_TEST_CASE(FoundationCreationConcativeOutline)
+{
+    Mesh mesh;
+    mesh.read("../test_models/concave_outline_binary.stl");
+    Mesh output_mesh = createFoundation(mesh, 7.00, 5.00);
+    auto facets = output_mesh.getFacets();
+    int i = 0, j = facets.size() - 1;
+    TEST_ASSERT(facets[i].a.z - facets[j].a.z == 7.00, "Height of foundation must be 7.00");
+    output_mesh.writeBinary("../test_models/concative_foundation.stl");
```

Comment:
Those write operations in your test cases could be excluded, since they are redundant. Or at least clear all test artifacts at the end of the test case: `std::remove("path/to/temporary/file");`

Reply:
Ok, it will be better if I remove write operations, because we allready have unit tests for write method